src/main/scala/sbtavrohugger/FileWriter.scala

```diff
@@ -3,32 +3,50 @@ package sbtavrohugger;
 import avrohugger.Generator
 import java.io.File
 import sbt.Keys._
-import sbt.{ Logger, globFilter, singleFileFinder }
+import sbt.{Logger, globFilter, singleFileFinder}
+import sbt.Path._
 
 object FileWriter {
 
   private[sbtavrohugger] def generateCaseClasses(
     generator: Generator,
+    imports: Seq[File],
     srcDir: File,
     target: File,
     log: Logger): Set[java.io.File] = {
 
-    for (idl <- (srcDir ** "*.avdl").get) {
+    val importsGroupedByExtension = imports.groupBy(_.ext)
+
+    def getSchemasFor(extension: String): Seq[File] = {
+      val importsForExtension = importsGroupedByExtension.getOrElse(extension, Seq.empty)
+      val sortedSchemas = {
+        val allSchemas = (srcDir ** s"*.$extension").get
+        if (extension == "avsc") {
+          AVSCFileSorter.sortSchemaFiles(allSchemas)
```

Comment:
For what it's worth, I don't really understand what [this call](https://github.com/julianpeeters/sbt-avrohugger/blob/master/src/main/scala/sbtavrohugger/AVSCFileSorter.scala#L12) is really doing. It seems to attempt to sort the schema files in such a way to avoid the problem I'm describing, but it doesn't seem to actually succeed in my use case. I'm practically brand new to Avro, so perhaps the schemas that were defined for me were defined incorrectly?

At any rate, I massaged the existing code such that it should all behave the same way as before, so I'm still invoking this call for all of the schema files not explicitly defined in the `imports` SBT Setting.